rapyuta_io_sdk_v2/client.py

```diff
@@ -187,3 +187,1012 @@ def get_project(self, project_guid: str = None) -> Munch:
+
+    def update_deployment(self, project_guid: str, name: str, body: dict) -> Munch:
+        """Update a deployment by its name.
+
+        Args:
+            name (str): Deployment name
+            body (dict): Update details
+
+        Returns:
+            Munch: Deployment details as a Munch object.
+        """
+        headers = self.config.get_headers()
+        if project_guid:
+            headers["project"] = project_guid
+
+        v2api_host = self.config.hosts.get("v2api_host")
+
+        response = self.c.put(
+            url=f"{v2api_host}/v2/deployments/{name}/",
+            headers=headers,
+            json=body,
+        )
+
+        handle_server_errors(response)
+
+        return munchify(response.json())
+
+    def delete_deployment(self, project_guid: str, name: str) -> Munch:
+        """Delete a deployment by its name.
+
+        Args:
+            name (str): Deployment name
+
+        Returns:
+            Munch: Deployment details as a Munch object.
+        """
+        headers = self.config.get_headers()
+        if project_guid:
+            headers["project"] = project_guid
+
+        v2api_host = self.config.hosts.get("v2api_host")
+
+        response = self.c.delete(
+            url=f"{v2api_host}/v2/deployments/{name}/",
+            headers=headers,
+        )
+
+        handle_server_errors(response)
+
+        return munchify(response.json())
+
+    def poll_deployment(
+        self,
+        project_guid: str,
+        name: str,
+        retry_count: int = 50,
+        sleep_interval: int = 6,
+        ready_phases: list[str] = None,
+    ) -> Munch:
+        """Poll a deployment by its name.
+
+        Args:
+            name (str): Deployment name
+            retry_count (int): Number of retries
+            sleep_interval (int): Sleep interval in seconds
+            ready_phases (list[str]): List of ready phases
+
+        Returns:
+            Munch: Deployment details as a Munch object.
+        """
+        pass
+
+    def stream_depolyment_logs(
+        self, project_guid: str, name: str, executable: str, replica: int = 0
+    ):
+        """Stream deployment logs.
+
+        Args:
+            name (str): Deployment name
+            executable (str): Executable name
+            replica (int): Replica number
+        """
+        pass
+
+    def list_disks(self, project_guid: str = None) -> Munch:
+        """List all disks in a project.
+
+        Returns:
+            Munch: List of disks as a Munch object.
+        """
+        headers = self.config.get_headers()
+        if project_guid:
+            headers["project"] = project_guid
+
+        v2api_host = self.config.hosts.get("v2api_host")
+
+        response = self.c.get(
+            url=f"{v2api_host}/v2/disks/",
+            headers=headers,
+        )
+
+        handle_server_errors(response)
+
+        return munchify(response.json())
+
+    def get_disk(self, project_guid: str, name: str) -> Munch:
```

Comment:
thought: Shouldn't the project_guid and organization_guid be coming from the context instead of accepting it explicitly in every method? Maybe we can support a **kwargs to override it but I think it should not be a mandatory parameter.